code/parachain/frame/composable-traits/src/staking/mod.rs

```diff
@@ -190,11 +192,12 @@ pub enum RewardPoolConfiguration<AccountId, AssetId, BlockNumber, RewardConfigs,
 /// should exist for each position when stored in the runtime storage.
 /// TODO refer to the relevant section in the design doc.
 #[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
-pub struct Stake<AccountId, RewardPoolId, Balance, Reductions> {
-	/// Protocol or the user account that owns this stake
-	// TODO (vim): Remove the owner and track the financial NFT ID. In order to prevent a direct
-	// dependency to NFTs we can also just use nft ID as position ID. 	pub financial_nft_id: ItemId
-	pub owner: AccountId,
+pub struct Stake<ItemId, RewardPoolId, Balance, Reductions> {
+	// REVIEW(benluelo): Are the following two fields necessary? They are redundant now with the
+	// new Stakes storage and fnft ids.
+	/// The NFT account that owns this stake. Used with the `AssetId` associated with the stake to
+	/// uniquely identify related nft.
+	pub financial_nft_id: ItemId,
```

Comment:
CollectionId != RewardPoolId
RewardPoolId == PoolAssetId

Also, we should rename `financial_nft_id` -> `fnft_collection_id` for clarity.

Reply:
6392c47b69812811dd786c24754de733845aca48